### PR TITLE
Fix compiler errors due to firrtl emitter api changes

### PR DESCRIPTION
**Pull request by chick**
Recent changes to emitter stuff in firrtl, broke the rather brittle verilator backend.  I have cleaned this up a fair amount, although there is probably still some room for improvement.
Reworked the VerilatorCppHarnessCompiler stuff.
It seemed to be unnecessarily using Transforms to do a simple template substitution.
chiselMain had it's own hooks into the above.  Cleaned this up as much as I dared
Fix deprecated code
Reg(init -> RegInit(
log2Up -> Log2Ceil
Fixed up style (I needed this to debug this mess)

### CODE REVIEW

**Comment by edwardcwang** [comment]
Seems to fix #122

**Comment by chick** [comment]
@edwardcwang Do you have time to review this PR, It's a little ugly because it includes style cleanup, but a lot of things are getting backed up behind this and other PR's are getting created that are fixing the same problem

File: src/test/scala/verilator/doohickey.scala [hunk 1]
```diff
@@ -2,10 +2,13 @@ package verilator
...
+/**
+  * The practice of creating Seq's or Vec's of IO's is problematic
+  */
 class doohickey() extends Module {
-  val io = new Bundle {
-  }
-  val bobs = Vec.fill(16) {
+  val io = IO(new Bundle {})
+  //TODO: figure out if this idiom is the key point here or if some other example would suffice
+  val bobs = Seq.fill(16) {
```
Comment (edwardcwang) [inline src/test/scala/verilator/doohickey.scala]: The ol' Veq vs Seq strikes again...

File: src/main/scala/chisel3/iotesters/PeekPokeTesterUtils.scala [hunk 2]
```diff
@@ -4,12 +4,12 @@ package chisel3.iotesters
 
 import java.io.File
 
-import chisel3.{Bundle, Data, Element, Module, Vec}
+import chisel3._
```
Comment (edwardcwang) [inline src/main/scala/chisel3/iotesters/PeekPokeTesterUtils.scala]: I suppose it's easier to import all of chisel3?

**Pull request merged**